sim/druid/insect_swarm.go

```diff
@@ -0,0 +1,70 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+func (druid *Druid) registerInsectSwarmSpell() {
+	hasGlyph := druid.HasPrimeGlyph(proto.DruidPrimeGlyph_GlyphOfInsectSwarm)
+
+	druid.InsectSwarm = druid.RegisterSpell(Humanoid|Moonkin, core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 5570},
+		SpellSchool:    core.SpellSchoolNature,
+		ProcMask:       core.ProcMaskSpellDamage,
+		ClassSpellMask: DruidSpellInsectSwarm | DruidNatureSpells | DruidSpellDoT,
+		Flags:          SpellFlagNaturesGrace | SpellFlagOmenTrigger | core.SpellFlagAPL,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost:   0.08,
+			Multiplier: 1,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+		},
+
+		DamageMultiplier: 1 + core.TernaryFloat64(hasGlyph, 0.3, 0),
+		ThreatMultiplier: 1,
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "Insect Swarm",
+			},
+
+			NumberOfTicks:       6,
+			TickLength:          time.Second * 2,
+			AffectedByCastSpeed: true,
+			BonusCoefficient:    0.13,
+
+			OnSnapshot: func(sim *core.Simulation, target *core.Unit, dot *core.Dot, isRollover bool) {
+				min, max := core.CalcScalingSpellEffectVarianceMinMax(proto.Class_ClassDruid, 0.138, 0)
+				baseDamage := sim.Roll(min, max)
+				dot.Snapshot(target, baseDamage)
```

Comment:
Are you expecting this to actually have variance and it just has 0 now in beta? If not you're doing random rolls for no reason

Reply:
changed to CalcScalingSpellAverageEffect()